facilitator/src/aggregation.rs

```diff
@@ -207,10 +219,37 @@ impl<'a> BatchAggregator<'a> {
                 Batch::new_validation(self.aggregation_name, batch_id, batch_date, !self.is_first),
                 &mut *self.peer_validation_transport.transport,
             );
-        let peer_validation_header = peer_validation_batch
-            .header(&self.peer_validation_transport.batch_signing_public_keys)?;
-        let own_validation_header = own_validation_batch
-            .header(&self.own_validation_transport.batch_signing_public_keys)?;
+
+        let peer_validation_header = match peer_validation_batch
+            .header(&self.peer_validation_transport.batch_signing_public_keys)
+        {
+            Ok(header) => header,
```

Comment:
nit: a bit of repeatedish code here between peer and own validation header, but I don't think it's really an issue.

Reply:
I agree. The tricky thing is that here, I want to propagate the error as `?` would, but I want to intercept it so I can increment the metric, and doing that is a bit awkward as we see here. I think a better solution would be to either:
- plumb the metric collector down into the place where the header was rejected (`batch.rs` in this case) and increment the counter there
- plumb a better error (i.e. a variant on `crate::Error` like `InvalidHeaderError`) all the way up to some context that has the metrics collector in scope and can increment the metric appropriately after matching on the error.

Both of these seem like more complicated exercises than I want to tackle in this PR.